public/javascripts/payments.js

```diff
@@ -283,156 +313,185 @@
     submitButton.disabled = true;
     submitButton.textContent = 'Processing…';
 
-    if (payment === 'card') {
-      // Let Stripe.js handle the confirmation of the PaymentIntent with the card Element.
-      const response = await stripe.confirmCardPayment(
-        paymentIntent.client_secret,
-        {
-          payment_method: {
-            card,
-            billing_details: {
-              name,
-              address: billingAddress,
-            },
-          },
-          shipping,
-        }
+    // Update Payment Intent if currency is different to default
+    if (config.currency !== activeCurrency) {
+      const response = await store.updatePaymentIntentWithCurrencyPaymentMethod(
+        paymentIntent.id,
+        activeCurrency,
+        [payment]
       );
-      handlePayment(response);
-    } else if (payment === 'sepa_debit') {
-      // Confirm the PaymentIntent with the IBAN Element.
-      const response = await stripe.confirmSepaDebitPayment(
-        paymentIntent.client_secret,
-        {
-          payment_method: {
-            sepa_debit: iban,
-            billing_details: {
-              name,
-              email,
+
+      if (response.error) {
+        handleError(response);
+        updateError = true;
+      }
+    }
+    if (!updateError) {
```

Comment:
I've removed the updateError variable and check all together, rather I've just returned out if there's an error (so that the rest of the code isn't executed). Would this be suitable?